### ISSUE TITLE
[RFC] Input type file support

### ISSUE THREAD

**Issue by italoiz** [Rocketseat/unform#81]
I noticed that the `Input` component does not register the values for `type="file"`. I think it would be interesting to add this feature to the library.

With that in mind, I thought of two ways to do this:

- Change the `Input` component to handle the` onChange` property.
- Create a new `FileInput` component just for this purpose.

Using any of the ways suggested above, we would have an answer like:

```json
{
  "attach": "data:text/csv;base64..."
}
```

**Examples**

Creating a new `FileInput` component would normally be like the other components:

```jsx
<Form>
  <FileInput name="attach" />
</Form>
```

Using the `Input` component:

```jsx
<Form>
  <Input name="attach" type="file" />
</Form>
```

**Comment by diego3g** [comment]
Maybe instead of returning a base64 we could provide an way to return a [FormData](https://developer.mozilla.org/pt-BR/docs/Web/API/FormData) instead of an object in `handleSubmit` method as we already have full control over the object generated in [parseFormData](https://github.com/Rocketseat/unform/blob/master/lib/Form.tsx#L46) function.

The downside of this implementation is that the backend must support urlencoded form data instead of only JSON.

**Comment by italoiz** [comment]
What if we were given the option to set the return type?

Example:

```jsx
<Form returnType="json | formData">
  <Input name="name" />
  <FileInput name="attach" />
</Form>
```

**Comment by diego3g** [comment]
@italoiz Great! I think it's a start. Can you try to create a PR? Maybe we can wait more contributions too.

**Comment by italoiz** [comment]
Good @diego3g, I'll do the PR today.

**Comment by italoiz** [comment]
@diego3g 

I have found problems to use the suggested solution using `FormData`, because in cases of nested properties like `<Scope path =" ... ">` does not work, because "FormData" does not accept values as objects.

When using "FormData" this is given as values, eg:
```
people [object Object],[object Object]
```

Are there any suggestions?

**Comment by diego3g** [comment]
@italoiz Can you try using the base64 first and upload some image to a backend server to test it? I read about base64 and multipart form-data and i think we should use base64 only for now, the only limitations related to this type of upload is the size of upload, so we won't have problems with small uploads. 

I also think that there are other approaches the user might like to use to upload images without unform itself. For example, today in all of my applications i have an upload route where i upload all my files as soon as the user select it from the input, and then when the user send the main form that uses that image i only send the image ID that is already stored in the backend, not the image itself.

I think that if we provide an easy way to send base64 it's great.

**Comment by pellizzetti** [comment]
Closed in #82
